File: v2/frontend/types/Corpus.ts
export interface Corpus {
  name: string;
  description?: string | null;
  id: string;
  status: string;
  created_by_id: string;
  created_at: string;
  updated_at: string;
}

export interface CorpusCreate {
  name: string;
  description?: string | null;
}

export interface CorpusUpdate {
  name: string;
  description?: string | null;
}

File: v2/app/schemas/Corpus.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel

class CorpusBase(BaseModel):
    name: str
    description: Optional[str] = None

class CorpusCreate(CorpusBase):
    pass

class CorpusUpdate(CorpusBase):
    pass

class CorpusInDBBase(CorpusBase):
    id: str
    status: str
    created_by_id: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class Corpus(CorpusInDBBase):
    pass
